capstone-app/src/app/modules/photos/components/photos-section/photos-section.component.ts

```diff
@@ -93,9 +112,12 @@ export class PhotosSectionComponent implements OnChanges {
   }
 
   extractUniqueTypes() {
-    this.allTypes = Object.keys(this.mockPlaces)
-      .map((key) => this.mockPlaces[key].types)
-      .flat();
+    this.allTypes = [];
+    const tempTypes: [string[]] = [[]];
+    for (const [key, value] of this.markerPlaces) {
+      tempTypes.push(value);
+    }
+    this.allTypes = tempTypes.flat();
     const copyForSplicing = this.allTypes.slice();
     this.uniqueTypes = this.allTypes = this.allTypes.filter((category) => {
```

Comment:
I didn't even realize this worked. Can we split this into two different assignment lines?

Reply:
Sorry, a little confused by what you mean by this?